### ISSUE TITLE
Windows: amethyst_assets (font) loader does not work with mixed slashes

### ISSUE THREAD

**Issue by azriel91** [amethyst/amethyst#622]
Hiya, I found a bug when loading fonts on Windows. If I instantiate the Amethyst application with an assets directory path like `"assets"`, and load a Font asset using "relative/forward_slashed/path/to/font.ttf", it works.

However, if I instantiate it with a canonicalized path, like:

```rust
// pseudo code
let assets_dir = env::current_exe()?
    .parent()?
    .canonicalize()?
    .join("assets");
let application = Application::new(assets_dir, /* .. */) // .. bundles etc.
```

Then when I go `loader.load("forward_slashed/path/to/font.ttf", /* .. */);`, it fails the load the path.
If I use `loader.load("back_slashed\\path\\to\\font.ttf", /* .. */);`, it works.

The actual log output I get is as follows:

```
[amethyst_assets::loader][DEBUG] "ui::Font": Loading asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf" with format "TTF/OTF" from source "[default source]" (handle id: Handle { id: 0, marker: PhantomData })
[amethyst_assets::loader][DEBUG] "ui::Font": Loading asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Bold.ttf" with format "TTF/OTF" from source "[default source]" (handle id: Handle { id: 1, marker: PhantomData })
[amethyst_assets::loader][DEBUG] "ui::Font": Loading asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-It.ttf" with format "TTF/OTF" from source "[default source]" (handle id: Handle { id: 2, marker: PhantomData })
[amethyst_assets::loader][DEBUG] "ui::Font": Loading asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-BoldIt.ttf" with format "TTF/OTF" from source "[default source]" (handle id: Handle { id: 3, marker: PhantomData })
[amethyst_assets::storage][ERROR] "ui::Font": Asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf" (handle id: Handle { id: 0, marker: PhantomData }) could not be loaded: Failed to load asset with name "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf"
[amethyst_assets::progress][ERROR] error: Failed to load asset with name "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf"
[amethyst_assets::progress][ERROR] caused by: Format "TTF/OTF" could not load asset
[amethyst_assets::progress][ERROR] caused by: Failed to load bytes from source
[amethyst_assets::progress][ERROR] caused by: Failed to fetch metadata for "\\\\?\\D:\\work\\gitlab\\azriel91\\autexousious\\assets\\font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf"
[amethyst_assets::progress][ERROR] caused by: The filename, directory name, or volume label syntax is incorrect. (os error 123)
[amethyst_assets::progress][ERROR] note: to handle the error, use a `Progress` other than `()`
[amethyst_assets::storage][ERROR] "ui::Font": Asset "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Bold.ttf" (handle id: Handle { id: 1, marker: PhantomData }) could not be loaded: Failed to load asset with name "font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Bold.ttf"

// ... more of the same type of error
```

The important line is:

```
[amethyst_assets::progress][ERROR] caused by: Failed to fetch metadata for "\\\\?\\D:\\work\\gitlab\\azriel91\\autexousious\\assets\\font/source-code-pro-2.030R-ro-1.050R-it/TTF/SourceCodePro-Regular.ttf"
```

I haven't yet dived into the asset loader code to check where this could be fixed.

**Comment by Rhuagh** [comment]
Don't think rust supports mixing path delimiters

**Comment by torkleyy** [comment]
We can just replace all backslashes when we receive the assets directory.
For all the other paths, only forward is allowed.

**Comment by Rhuagh** [comment]
Why not just store a PathBuf or Path?

**Comment by Rhuagh** [comment]
We should be able to ignore the delimiter by using the APIs on Path/PathBuf tbh.

**Comment by azriel91** [comment]
Here's where the asset storage location and the asset path are combined: [amethyst_assets/src/source/dir.rs#L28-L33](https://github.com/amethyst/amethyst/blob/develop/amethyst_assets/src/source/dir.rs#L28-L33).

Looks like Rust's `Path.push` doesn't automatically correct the slashes 😞. I have to go now, but if someone hasn't gotten to it before my tomorrow then I'll experiment with potential fixes.

**Comment by torkleyy** [comment]
While I think we should recommend only using forward slashes for asset identifiers, I'm fine with using `Path::from` on the asset identifier.

**Comment by azriel91** [comment]
Fixed in #623
